lib/Controller/FaceController.php

```diff
@@ -81,4 +122,82 @@ public function getThumb ($id, $size) {
 		return $resp;
 	}
 
+	/** @scrutinizer ignore-unused */
+	private function hipsterize(&$image, &$face) {
+		$imgResource = $image->resource();
+
+		$landmarks = json_decode($face->getLandmarks(), true);
+		if (count($landmarks) === 5) {
+			$eyesX1 = $landmarks[2]['x'];
+			$eyesY1 = $landmarks[2]['y'];
+
+			$eyesX2 = $landmarks[0]['x'];
+			$eyesY2 = $landmarks[0]['y'];
+
+			$eyesXC = ($eyesX2 + $eyesX1)/2;
+			$eyesYC = ($eyesY2 + $eyesY1)/2;
+
+			$mustacheXC = $landmarks[4]['x'];
+			$mustacheYC = $landmarks[4]['y'];
+		}
+		else if (count($landmarks) === 68) {
+			$eyesX1 = $landmarks[36]['x'];
+			$eyesY1 = $landmarks[36]['y'];
+			$eyesX2 = $landmarks[45]['x'];
+			$eyesY2 = $landmarks[45]['y'];
+
+			$eyesXC = ($eyesX2 + $eyesX1)/2;
+			$eyesYC = ($eyesY2 + $eyesY1)/2;
+
+			$mustacheXC = $landmarks[52]['x'];
+			$mustacheYC = $landmarks[52]['y'];
+		}
+		else {
+			return;
+		}
+
+		$eyesW = $eyesX2 - $eyesX1;
+		$eyesH = $eyesY2 - $eyesY1;
+
+		$eyesL = sqrt(pow($eyesW, 2) + pow($eyesH, 2));
+		$angle = rad2deg(atan(-$eyesH/$eyesW));
+
+		$glassesGd = imagecreatefrompng(\OC_App::getAppPath('facerecognition') . '/img/glasses.png');
```

Comment:
should you mention somewhere is this image CC? do you need to mention attribution?

Reply:
Maybe .. But where? The application icons also not are licensed..

Reply:
p.s: They are images originally with CC, that I modify to adapt, and they are still CC obviously.